packages/platform-ui/src/components/RunEventDetails.tsx

```
@@ -29,6 +29,85 @@ const safeJsonParse = (value: string): unknown => {
   }
 };
 
+const extractReasoningMetrics = (value: unknown): { tokens?: number; score?: number } => {
+  const visited = new WeakSet<object>();
+
+  const walk = (current: unknown): { tokens?: number; score?: number } => {
+    if (current === null || current === undefined) {
+      return {};
+    }
+
+    if (typeof current === 'number') {
+      return { tokens: current };
+    }
+
+    if (typeof current === 'string') {
+      const numeric = Number(current);
+      if (!Number.isNaN(numeric)) {
+        return { tokens: numeric };
+      }
+      const parsed = safeJsonParse(current);
+      if (parsed !== current) {
+        return walk(parsed);
+      }
+      return {};
+    }
+
+    if (typeof current !== 'object') {
+      return {};
+    }
+
+    if (Array.isArray(current)) {
+      for (const item of current) {
+        const metrics = walk(item);
+        if (metrics.tokens !== undefined || metrics.score !== undefined) {
+          return metrics;
+        }
+      }
+      return {};
+    }
+
+    const record = current as Record<string, unknown>;
+    if (visited.has(record)) {
+      return {};
+    }
+    visited.add(record);
+
+    const directTokens =
+      asNumber(record.tokens) ??
+      asNumber(record.reasoningTokens) ??
+      asNumber(record.reasoning_tokens) ??
+      asNumber(record.token_count) ??
+      asNumber(record.totalTokens) ??
+      asNumber(record.total) ??
+      asNumber(record.value) ??
+      asNumber(record.count);
+
+    const directScore =
+      asNumber(record.score) ??
+      asNumber(record.reasoningScore) ??
+      asNumber(record.confidence);
+
+    if (directTokens !== undefined || directScore !== undefined) {
```

[major] This helper returns as soon as it finds any numeric `score`, even if the same object still contains nested metrics with the reasoning token count. For provider payloads shaped like `{ score: 0.42, metrics: { tokens: 120 } }`, we bail out before we recurse into `metrics`, so the token count is silently dropped. Please keep searching nested keys when only a score is present (or defer the early return until we actually find tokens) so we don’t hide valid reasoning-token data.

---

Updated `extractReasoningMetrics` so we keep traversing nested keys when only a score is present. The helper now aggregates tokens/score separately before returning, which preserves payloads shaped like `{ score: 0.42, metrics: { tokens: 120 } }`.